lexers/autohotkey.lua

```diff
@@ -0,0 +1,665 @@
+-- AutoHotkey LPeg lexer.
+
+local lexer = lexer
+local P, S = lpeg.P, lpeg.S
+
+local lex = lexer.new(...)
+
+-- Keywords.
+lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD, true)))
+
+-- Variables.
+lex:add_rule('variable_builtin', lex:tag(lexer.VARIABLE_BUILTIN, 'A_' * lex:word_match(lexer.VARIABLE_BUILTIN, true)))
+
+-- Constants.
+lex:add_rule('constant', lex:tag(lexer.CONSTANT_BUILTIN, S('fF') * lexer.digit * (lexer.digit)^-1
+    + lex:word_match(lexer.CONSTANT_BUILTIN, true)))
+
+-- Functions.
+lex:add_rule('function_builtin',
+  lex:tag(lexer.FUNCTION_BUILTIN, lex:word_match(lexer.FUNCTION_BUILTIN, true)) *
+    #(lexer.space^0 * P('(')^-1))
+local func = lex:tag(lexer.FUNCTION, lexer.word)
+local method = lpeg.B('.') * lex:tag(lexer.FUNCTION_METHOD, lexer.word)
+lex:add_rule('function', (method + func) * #(lexer.space^0 * '('))
+
+-- Identifiers.
+lex:add_rule('identifier', lex:tag(lexer.IDENTIFIER, lexer.word))
+
+-- Comments.
+local line_comment = lexer.to_eol(';')
+local block_comment = lexer.range('/*', '*/')
+lex:add_rule('comment', lex:tag(lexer.COMMENT, line_comment + block_comment))
+
+-- Preprocessor.
+lex:add_rule('preprocessor', lex:tag(lexer.PREPROCESSOR, '#' * lex:word_match(lexer.PREPROCESSOR, true)))
+
+-- Strings.
+local dq_str = lexer.range('"', true, false)
+local sq_str = lexer.range("'", true, false)
+lex:add_rule('string', lex:tag(lexer.STRING, dq_str + sq_str))
+
+-- Numbers.
+lex:add_rule('number', lex:tag(lexer.NUMBER, lexer.number))
+
+-- Operators.
+lex:add_rule('operator', lex:tag(lexer.OPERATOR, S('~+-^*/&<>=?:()[]{}')))
+
+lex:set_word_list(lexer.KEYWORD, {
+  'as', 'and', 'contains', 'false', 'in', 'is', 'IsSet', 'not', 'or', 'super', 'true', 'unset',
+  'Break', 'Catch', 'Continue', 'Else', 'Finally', 'For', 'Global', 'Goto', 'If', 'Local', 'Loop', 'Return', 'Static', 'Throw', 'Try', 'Until', 'While'
+})
+
+lex:set_word_list(lexer.FUNCTION_BUILTIN, {
```

Comment:
When I commit this, is it okay if I have my Lua code formatter format it like your keyword list? If you want to separate words into sections, you can use line comments like in https://github.com/orbitalquark/scintillua/blob/3559ca9a7859a69385f94ce596233c9a1d582fc2/lexers/ansi_c.lua#L98

Reply:
Absolutely, in fact I was relying on you to format this properly. It's only in this format because that's how it ended up after I pulled the data from the documentation and I don't currently have a lua formatter set up.